Import libraries

In [1]:
import requests
from bs4 import BeautifulSoup

In [25]:
# Fetch the Wikipedia page
url = "https://en.wikipedia.org/wiki/DeepSeek_(chatbot)"
response = requests.get(url)

In [27]:
soup = BeautifulSoup(response.content, "html.parser")
print (soup)

<!DOCTYPE html>

<html class="client-nojs vector-feature-language-in-header-enabled vector-feature-language-in-main-page-header-disabled vector-feature-page-tools-pinned-disabled vector-feature-toc-pinned-clientpref-1 vector-feature-main-menu-pinned-disabled vector-feature-limited-width-clientpref-1 vector-feature-limited-width-content-enabled vector-feature-custom-font-size-clientpref-1 vector-feature-appearance-pinned-clientpref-1 vector-feature-night-mode-enabled skin-theme-clientpref-day vector-sticky-header-enabled vector-toc-available" dir="ltr" lang="en">
<head>
<meta charset="utf-8"/>
<title>DeepSeek (chatbot) - Wikipedia</title>
<script>(function(){var className="client-js vector-feature-language-in-header-enabled vector-feature-language-in-main-page-header-disabled vector-feature-page-tools-pinned-disabled vector-feature-toc-pinned-clientpref-1 vector-feature-main-menu-pinned-disabled vector-feature-limited-width-clientpref-1 vector-feature-limited-width-content-enabled vecto

Parsing

In [29]:
soup = BeautifulSoup(response.content, "html.parser")

In [31]:
title = soup.find("h1").text
print("Page Title:", title)

Page Title: DeepSeek (chatbot)


In [33]:
intro_div = soup.find("div", class_="mw-parser-output")
intro_paragraphs = []
if intro_div:
    for tag in intro_div.find_all("p", recursive=False):
        text = tag.get_text(strip=True)
        if text:
            intro_paragraphs.append(text)
        if len(intro_paragraphs) >= 2:
            break
introduction = "\n\n".join(intro_paragraphs) if intro_paragraphs else "No Introduction"
print("\nIntroduction:\n", introduction)


Introduction:
 DeepSeek[a]is achatbotcreated by the Chinese artificial intelligence companyDeepSeek.

Released on 10 January 2025,DeepSeek-R1surpassedChatGPTas the most downloadedfreewareapp on theiOS App Storein the United States by 27 January.[1]DeepSeek's success against larger and more established rivals has been described as "upending AI"[1]and initiating "a global AIspace race".[2]DeepSeek's compliance with Chinese governmentcensorship policiesand itsdata collection practiceshave also raised concerns over privacy and information control in the model, prompting regulatory scrutiny in multiple countries. However, it has also been praised for its open weights and infrastructure code, energy efficiency and contributions to theopen-source community.[3][4]


 Extract Section Titles and First Paragraphs

In [35]:
sections = soup.find_all("h2")

for section in sections:
    section_title = section.text.strip().replace("[edit]", "")
    section_content_tag = section.find_next("p")
    if section_content_tag:
        section_content = section_content_tag.text.strip()
        print(f"Section Title: {section_title}")
        print(f"Section Content: {section_content[:300]}...")
        print()

Section Title: Contents
Section Content: ...

Section Title: History
Section Content: On 10 January 2025, DeepSeek released the chatbot, based on the DeepSeek-R1 model, for iOS and Android. By 27 January, DeepSeek-R1 surpassed ChatGPT as the most-downloaded freeware app on the iOS App Store in the United States,[1] which resulted in an 18% drop in Nvidia’s share price.[5][6] After a ...

Section Title: Usage
Section Content: DeepSeek can answer questions, solve logic problems, and write computer programs on par with other chatbots, according to benchmark tests used by American AI companies.[1]...

Section Title: Operation
Section Content: DeepSeek-V3 uses significantly fewer resources compared to its peers. For example, whereas the world's leading AI companies train their chatbots with supercomputers using as many as 16,000 graphics processing units (GPUs), DeepSeek claims to have needed only about 2,000 GPUs—namely, the H800 series ...

Section Title: Reactions
Section Content: DeepSe

Fetch the section History

In [49]:
headers = soup.find_all(['h2', 'h3'])

# Iterate through headers to find the "History" section
history_section_found = False
for header in headers:
    if header.get_text(strip=True).lower() == 'history':
        history_section_found = True
        # Start extracting paragraphs after the "History" header
        next_node = header.find_next()
        while next_node and next_node.name not in ['h2', 'h3']:
            if next_node.name == 'p':
                print(next_node.get_text())
            next_node = next_node.find_next()

if not history_section_found:
    print("History section not found.")

On 10 January 2025, DeepSeek released the chatbot, based on the DeepSeek-R1 model, for iOS and Android. By 27 January, DeepSeek-R1 surpassed ChatGPT as the most-downloaded freeware app on the iOS App Store in the United States,[1] which resulted in an 18% drop in Nvidia’s share price.[5][6] After a "large-scale" cyberattack on the same day disrupted the proper functioning of its servers, DeepSeek limited its new user registration to phone numbers from mainland China, email addresses, or Google account logins.[7][8]

On 3 April 2025, in collaboration with researchers at Tsinghua University, DeepSeek published a paper unveiling a new model that combines the techniques generative reward modeling (GRM) and self-principled critique tuning (SPCT). The resulting model is referred to as DeepSeek-GRM. The goal of using these techniques is to foster more effective inference-time scaling within their LLM and chatbot services. Notably DeepSeek has said that these new models will be released and ma

Extract the links

In [53]:
external_links = soup.find_all("a", {"class": "external text"})
print("External Links:")
for link in external_links:
    href = link.get("href")
    if href:
        print(href)

External Links:
https://chat.deepseek.com/
https://www.nytimes.com/2025/01/23/technology/deepseek-china-ai-chips.html
https://search.worldcat.org/issn/0362-4331
https://web.archive.org/web/20250123102900/https://www.nytimes.com/2025/01/23/technology/deepseek-china-ai-chips.html
https://abcnews.go.com/Business/nvidia-microsoft-shares-tumble-china-based-ai-app/story?id=118136157
https://www.cnbc.com/2025/01/27/chinas-deepseek-ai-tops-chatgpt-app-store-what-you-should-know.html
https://web.archive.org/web/20250128201052/https://www.cnbc.com/2025/01/27/chinas-deepseek-ai-tops-chatgpt-app-store-what-you-should-know.html
https://www.cbsnews.com/news/what-is-deepseek-ai-china-stock-nvidia-nvda-asml/
https://web.archive.org/web/20250129002522/https://www.cbsnews.com/news/what-is-deepseek-ai-china-stock-nvidia-nvda-asml/
https://www.theguardian.com/technology/2025/jan/27/deepseek-cyberattack-ai
https://www.businessinsider.com/deepseek-issues-account-registration-limited-china-phone-numbers-mali

References

In [55]:
references = []
for ref in soup.find_all("cite"):
    link = ref.find("a", href=True)
    if link:
        references.append(link['href'])
print("\nReferences:")
for ref in references:
    print(ref)


References:
https://www.nytimes.com/2025/01/23/technology/deepseek-china-ai-chips.html
https://abcnews.go.com/Business/nvidia-microsoft-shares-tumble-china-based-ai-app/story?id=118136157
https://www.cnbc.com/2025/01/27/chinas-deepseek-ai-tops-chatgpt-app-store-what-you-should-know.html
https://www.cbsnews.com/news/what-is-deepseek-ai-china-stock-nvidia-nvda-asml/
https://www.theguardian.com/technology/2025/jan/27/deepseek-cyberattack-ai
https://www.businessinsider.com/deepseek-issues-account-registration-limited-china-phone-numbers-malicious-attack-2025-1
https://www.scmp.com/tech/tech-trends/article/3305259/deepseek-unveils-new-ai-reasoning-method-anticipation-its-next-gen-model-rises?module=perpetual_scroll_0&pgtype=article
/wiki/ArXiv_(identifier)
https://techcrunch.com/2025/04/30/deepseek-upgrades-its-ai-model-for-math-problem-solving/
https://www.scmp.com/tech/tech-trends/article/3308566/deepseek-quietly-updates-open-source-model-handles-maths-proofs
https://economictimes.indiat

Sentiment analysis 

In [73]:
from textblob import TextBlob

In [75]:
# Get the page text
page_text = soup.get_text()

# Step 3: Clean up the extracted text (optional)
# Remove extra newlines and spaces
page_text = ' '.join(page_text.split())

# Step 4: Perform Sentiment Analysis with TextBlob
blob = TextBlob(page_text)

# Step 5: Output the sentiment analysis results
print("Sentiment Analysis Result:")
print(f"Polarity: {blob.sentiment.polarity}")  # Range: -1 (negative) to 1 (positive)
print(f"Subjectivity: {blob.sentiment.subjectivity}")  # Range: 0 (objective) to 1 (subjective)

Sentiment Analysis Result:
Polarity: 0.09721527198591419
Subjectivity: 0.3768874856030817


Polarity: 0.097 — The sentiment of the entire article is slightly positive, but not very strong. A polarity close to 0 means it's fairly neutral, with a slight tendency toward positive sentiment.

Subjectivity: 0.377 — The article is somewhat objective (close to 0), with some level of subjectivity, as expected for a Wikipedia page. This indicates that the article contains a mix of factual content and potentially opinion-based information.